metagraph/src/graph/alignment/alignment.cpp

```diff
@@ -796,6 +1226,9 @@ bool spell_path(const DeBruijnGraph &graph,
 }
 
 bool Alignment::is_valid(const DeBruijnGraph &graph, const DBGAlignerConfig *config) const {
+    if (empty())
+        return true;
+
     std::string path;
     if (!spell_path(graph, nodes_, path, offset_)) {
```

Comment:
Why do you reconstruct the seq from the path if you could save it beforehand?

Can you add comments to describe what `spell_path` returns (when true and when false)?

Reply:
The whole point of this function is to check if the path stored in the alignment spells the stored sequence.

Reply:
...and when an alignment is loaded from a json

Reply:
Also, wouldn't it be easier to just call map_to_nodes_sequentially instead of doing it the other way around?

Reply:
Since alignments which come from post-alignment chaining could have gaps (i.e., `nodes_` contains `npos`), this needs a special function to spell those kinds of paths.

I refactored this so that `spell_path` returns the spelling, but throws runtime errors if it encounters anything weird. `is_valid` then tries to catch these.

Reply:
Yeah, but still you could just call map_to_nodes + some postprocessing. But probably this way is also fine.